exchange_calendars/exchange_calendar.py

```diff
@@ -250,13 +333,21 @@ def __init__(self, start: Date | None = None, end: Date | None = None):
         self.last_trading_session = _all_days[-1]
 
         self._late_opens = pd.DatetimeIndex(
-            _special_opens.map(self.minute_index_to_session_labels)
+            _special_opens.map(lambda x: self.minute_index_to_session_labels(x, "both"))
```

Comment:
Why is this set to both?

Reply:
It shouldn't be! `_special_opens` should map to sessions based on minutes evaluated for side "left" and `_special_closes` should map to sessions based on minutes evaluated for side "right".  It could be left as "both" if not for the 24 hour calendars (which don't have a "both" option). I had it as "both" simply to maintain the original behaviour - the only reason this didn't result in a bug is that no existing 24 hour calendar has a late open. I would now change these to "left" and "right", but...

Having looked into what's going on here the whole mapping appears to be superfluous. `_late_opens` as evaluated is actually already represented by `_special_opens.index` and `_early_closes` by `_special_closes.index`, i.e. the assignment line under review could be replaced with a simple:
```python
self._late_opens = _special_opens.index
```
I'm not sure where the need for the method call came from? Perhaps `_special_opens` was originally defined differently? In any event, I'll add a commit to change these assignments to the direct assignment (as above). Let me know if you think the mapping is necessary and that I'm missing something here.

NB a consequence of losing the call to `minute_index_to_session_labels` is that it's more likely that the `*_minute_nanos` methods will end up being changed to properties (item 5 on #86).